src/main/java/org/icgc/argo/workflow_management/rabbitmq/ExecuteConsumerConfig.java

```diff
@@ -56,43 +60,16 @@
   private final WorkflowExecutionService wes;
   private final RabbitEndpointService rabbit;
 
-  @Autowired
-  public ExecuteConsumerConfig(
-      WorkflowExecutionService wes,
-      RabbitEndpointService rabbit,
-      WebLogEventSender webLogEventSender) {
-    this.wes = wes;
-    this.rabbit = rabbit;
-    this.webLogEventSender = webLogEventSender;
-  }
-
   @Bean
   public Disposable wfMgmtRunMsgForExecuteConsumer() {
-    val dlxName = topicExchangeName + "-dlx";
-    val dlqName = queueName + "-dlq";
-    return rabbit
-        .declareTopology(
-            topologyBuilder ->
-                topologyBuilder
-                    .declareExchange(dlxName)
-                    .and()
-                    .declareQueue(dlqName)
-                    .boundTo(dlxName)
-                    .and()
-                    .declareExchange(topicExchangeName)
-                    .type(ExchangeType.topic)
-                    .and()
-                    .declareQueue(queueName)
-                    .boundTo(topicExchangeName, topicRoutingKeys)
-                    .withDeadLetterExchange(dlxName))
-        .createTransactionalConsumerStream(queueName, WfMgmtRunMsg.class)
+    return createTransConsumerStream(rabbit, topicExchangeName, queueName, topicRoutingKeys)
```

Comment:
So i'm a big fan of assigning the disposable to a state variable in the enclosing class rather than letting be a free floating bean. This way we can provide an API endpoint to check the status of the disposables in case they ever die.

Reply:
That's a good idea will update :+1:  here and in the other configs